packages/language-server/src/plugins/typescript/features/FindComponentUsagesProvider.ts

```diff
@@ -0,0 +1,127 @@
+import { Location, Position, Range } from 'vscode-languageserver';
+import { URI } from 'vscode-uri';
+import { flatten, pathToUrl } from '../../../utils';
+import { FindComponentUsagesProvider } from '../../interfaces';
+import { LSAndTSDocResolver } from '../LSAndTSDocResolver';
+import { convertToLocationRange, hasNonZeroRange } from '../utils';
+import { Document, DocumentManager } from '../../../lib/documents';
+import ts from 'typescript';
+import { isNoTextSpanInGeneratedCode, SnapshotFragmentMap } from './utils';
+import { SvelteSnapshotFragment } from '../DocumentSnapshot';
+import { lsConfig } from '../../../ls-config';
+export class FindComponentUsagesProviderImpl implements FindComponentUsagesProvider {
+    constructor(private readonly lsAndTsDocResolver: LSAndTSDocResolver) {}
+
+    async findComponentUsages(uri: string): Promise<Location[] | null> {
+        const u = URI.parse(uri);
+        const fileName = u.fsPath;
+        const document = await this.getDocument(fileName);
```

Comment:
I think it would be better if resolving to a document would be done in PluginHost already, so you don't need to duplicate the logic here and so that the documents state is consistent.

Reply:
Is this what you had in mind? https://github.com/sveltejs/language-tools/pull/1509/commits/ddba005203791ef5f840848d26f645654c1f4d60

Reply:
> Is this what you had in mind? [ddba005](https://github.com/sveltejs/language-tools/commit/ddba005203791ef5f840848d26f645654c1f4d60)

Actually this is causing an issue because it would be pushing the getDocument all the way back to server.ts . I was originally modeling after fileReferences where it simply passes the URI around.

Reply:
Looked at this again and still having issues. Trying to use getDocument in pluginhost produce an error saying it was unopened, not sure what's going on or if this really needs to be changed